src/main/1.3/scala/com/holdenkarau/spark/testing/RDDComparisons.scala

```diff
@@ -26,13 +26,57 @@ import org.apache.spark.rdd._
 
 
 object RDDComparisons {
+  /**
+   * Compare two RDDs with order (e.g. [1,2,3] != [3,2,1])
+   * If the partitioners are not the same this requires multiple passes
+   * on the input.
+   * If they are equal returns None, otherwise returns Some with the first mismatch.
+   * If the lengths are not equal, one of the two components may be None.
+   */
+  def compareWithOrder[T: ClassTag](expected: RDD[T], result: RDD[T]): Option[(Option[T], Option[T])] = {
+    // If there is a known partitioner just zip
+    if (result.partitioner.map(_ == expected.partitioner.get).getOrElse(false)) {
+      compareWithOrderSamePartitioner(expected, result)
+    } else {
+      // Otherwise index every element
+      def indexRDD[T](rdd: RDD[T]): RDD[(Int, T)] = {
+        val counts = rdd.mapPartitions(itr => List(itr.size).toIterator).collect()
+        val indexed = rdd.mapPartitionsWithIndex{
+          case (idx, itr) =>
+            val start = if (idx == 0) 0 else counts(idx-1)
```

Comment:
Ah yes good catch. I'll switch our test to use 5 partitions so it catches this too.